### ISSUE TITLE
Plugin ignores webpack loaders (such as babel)

### ISSUE THREAD

**Issue by VaclavSynacek** [markdalgleish/static-site-generator-webpack-plugin#20]
If a project is using some webpack javascript loaders, such as babel for ES6 features, the output of these loaders is ignored by this plugin - instead original source files are executed (evaled) for the static page generation.

This results in strange non-isomorphic behavior when the same code runs fine on the client (because it is already processed by loaders such as babel transpiler) and fails during static site generation by this plugin (because the runtime engine does not support that ES6 feature).

In my example I was using babel to transpile from ES6 to standard JavaScript. I used this code in one react component:

```javascript
//find object on array of objects if it's property name is equal to local variable name
//this can be rewritten in hundred other ways without ES6, but that is not the point
var product = data[data.findIndex(p => p.name == name)];
```

The findIndex function is [ES6 new feature](http://babeljs.io/docs/learn-es2015/#math-number-string-object-apis) not interpreted by standard node 0.12.x. and most browsers. When the code was processed by webpack with the babel loader, the resulting transpiled code did run just fine in a browser. But when the same code is run by this plugin, it fails with an error and does not generate a static html page. The exact error message is `ERROR in undefined is not a function`.

In my case the workaround is using `node --harmony` to run webpack, because node 0.12.x in harmony mode just happens to supports this feature. But there are other ES6 features supported by babel, that might not get solved by `--harmony` flag. Also switching from node 0.12.x to 4.x might help with some features, but your mileage may vary (I have not tried this one, but 4.x supports greater ES6 feature set). Using polyfills is another option, but this kind of duplicates the usage of transpiler...

TL;DR; Please change this plugin implementation to use javascript as outputted by webpack, not as the original source to webpack.

**Comment by VaclavSynacek** [comment]
Confirmed that on nodejs 4.2.x the above example runs even without `--harmony`. However that still does not conceptually solve the usage of transpilers as webpack loaders.

**Comment by markdalgleish** [comment]
I don't think this is the case. I'm using JSX in my source files, which Node definitely doesn't understand. Do you have an working example somewhere so I can reproduce your issue?

**Comment by markdalgleish** [comment]
I'm closing this issue for now, please re-open if you can show me a working example.

**Comment by VaclavSynacek** [comment]
I'm sorry, I was wrong.

While trying to create simple working example I found the problem: I required some ES6 source files not only from the webpack entry script, but also directly from `webpack.config.js` (to generate the StaticSiteGeneratorPlugin routes from the same code as react-router routes). This of course makes the whole webpack fail with an ES6 execution errors if executed in vanilla `node`, but runs fine if executed in `babel-node` or `node -harmony` depending on the ES6 features used.

Consider this closed for ever.
